docs/learn/learn-staking-advanced.md

```diff
@@ -0,0 +1,385 @@
+session and utility pallets. This makes the stash account even more isolated than using a controller
+account since one can bond / unbond / bond more funds using the staking proxy account. However,
+it is important to remember that actions that can be performed by the proxy accounts are limited, and in the case of staking
+proxy, extrinsic calls to the balances pallet cannot be signed. This means it is not possible to do balance transfers
+on the proxied account through a staking proxy.
+
+## Bags List
+
+The nomination intents are placed in a so-called
+[bags-list](https://github.com/paritytech/substrate/pull/9507).
+{{ kusama: The bags list example below uses DOT for explaining the concepts. :kusama }}The bags list
+has two primary components, bags and nodes. The list is composed of bags that each describe a range
+of active bonded funds (e.g. the 1st bag will have nominators who staked 0 → 10 DOT, 2nd bag 11 → 20
+DOT, etc). Each bag contains nodes that correspond to nominators and their staked funds.
+
+The bags-list pallet is designed to be self-maintaining, with minimal effort from the blockchain,
+making it extremely scalable. Let us explore the sorting functionality of the bags list with an
+example. In the bags list below, there are 8 nodes (corresponding to 8 accounts with staked funds)
+placed in 3 bags. It can be observed that the list of nodes within the bags is arranged based on
+their insertion order and not based on the number of tokens bonded. For instance, the nodes in bag 1
+are arranged in this order: 15 → 12 → 19.
+
+![bags list example 1](../assets/staking/bags-list-example-1.png)
+
+Let's say the nominator with the stake of 19 DOT bonds 2 DOT additionally. This action would place
+that nominator node in bag 2, right after the node with 27 DOT.
+
+![bags list example 2](../assets/staking/bags-list-example-2.png)
+
+Once the nomination period ends, the NPoS election mechanism takes the nomination intents and their
+associated votes as input, and it outputs a set of validators. The bags are iterated from the most
+staked to the least staked. This could leave the last touched bag to only be partially iterated.
+This means that in some edge cases, the order of members within a bag is also important. Recall that
+within each bag, the iteration order is simply the insertion order. If only 7 nodes must be picked
+for the electing set, the nodes with 5 and 7 DOT will be selected while the node with 8 DOT will be
+left out.
+
+![bags list example 3](../assets/staking/bags-list-example-3.png)
+
+If one receives staking rewards and the amount of staked tokens within the stash account increases
+over time, the position within a bag changes and may also result in a change of bag. This may also
+happen if accounts within the bag bond more tokens or unbond tokens, one's account position and the
+position of other accounts in the bags list might change. These changes are not done automatically.
```

Comment:
, requiring the nominator to submit `rebag` extrinsics to update their position.

Reply:
Added